### ISSUE TITLE
"Invalid input syntax for integer" when updating alt and caption of image

### ISSUE THREAD

**Issue by MickyBizzle** [strapi/strapi#11455]
### "Invalid input syntax for type integer:" when updating alt and caption of image

When I try to add a caption and alt text to an uploaded image I get a server error with the strapi_administrator table passing an entire user object instead of just an ID.

Similar to [#5249](https://github.com/strapi/strapi/issues/5249) but solution doesn't work for me.

### Steps to reproduce the behavior

1. Strapi running in docker with Postgres
2. Upload an image
3. Add caption and alt text to image

### Expected behavior

Caption and alt text should be updated on image

### Code snippets

Error I get (with info sanitised):
```
error: select "strapi_administrator".* from "strapi_administrator" where "strapi_administrator"."id" = $1 limit $2 - invalid input syntax for type integer: "{"id":1,"firstname":"first_name","lastname":"last_name","username":null,"email":"my_email@email.com","password":"my_password","resetPasswordToken":null,"registrationToken":null,"isActive":true,"blocked":null,"preferedLanguage":null}"
```

### System

- Node.js version: 14.16.0
- NPM version: 6.14.11
- Strapi version: 3.6.8 CE
- Database: Postgres
- Operating system: Docker

**Comment by derrickmehaffy** [comment]
Can reproduce on v3.6.8 or beta.11

**Comment by thomasgrivet** [comment]
Hello, we got the same error here with the same stack

**Comment by derrickmehaffy** [comment]
If it can't be reproduced :shrug: Setting as closed until someone else can give more information to reproduce.

**Comment by lukearmstrong** [comment]
I know this isn't a very useful contribution, but also getting this error with this stack.

**Comment by egtoney** [comment]
@derrickmehaffy I have some more info on this bug. Figured I'd post it here instead of starting a new issue for the same problem.

It seems the error is tied to the use of PSQL. My dev environment using SQLite does not run into the error.

It is only thrown when performing a PUT or POST on an object with relations. These relations must also be specified with the whole object, not just the id. For example:

This would throw an error
```
{
   "field": "example",
   "user": {
      "id": 1,
      "email": "test@example.com"
   }
}
```

While this would not
```
{
   "field": "example",
   "user": 1
}
```

It seems the error is thrown when the internal logic pulls the related object (I assume to check and see if an update is necessary). The code doesn't use the `id` field, and instead passes the whole object as a string to the database engine causing the parse error listed above. Let me know if you need any more info. I've done a ton of digging through the strapi source code and my specific situation but haven't found anything yet.

**Comment by lukearmstrong** [comment]
We discovered that we had previously overridden the file/controller by copying it to our application, but we did this on an older version and didn't update the other functions in this file so it was out of date.

So we copied the latest version of the file, removed the functions we didn't need to override, then made changes to the functions we wanted to override, and now it works.
